tez-dag/src/main/java/org/apache/tez/dag/app/rm/YarnTaskSchedulerService.java

```diff
@@ -2163,6 +2166,17 @@ void addDelayedContainer(Container container,
       }
     }
 
+    void removeDelayedContainer(HeldContainer container) {
+      if (container != null) {
+        synchronized(this) {
+          if (delayedContainers.remove(container)) {
+            LOG.info("Removed {} from delayed containers", container.getContainer().getId());
+          } else {
+            LOG.warn("Unknown container {} sent for removal. Ignoring.", container.getContainer().getId());
```

Comment:
A valid case here is when a new container is allocated - added to delayedContainers, it is polled (removed) from queue but if there is no pending task request `releaseContainer()` is invoked. Please suggest if the log level should be changed in `info` or for both the newly added logs should be at `debug`.

Reply:
nit: I think we should change the log level to debug and print the only if LOG.isDebugEnabled() is true, 

Similar to : 
https://github.com/apache/tez/pull/374/files#diff-aaf0735de6615f8eeb8469ceadca2c3b0cdd578544a2564c37b299c55347188eR1548-R1550

Rest of the code LGTM .

Reply:
not sure about the log levels, let's keep the simple remove path on debug:
```
            LOG.debug("Removed {} from delayed containers", container.getContainer().getId());
```
isDebugEnabled is not necessary as long as we use the {} formatting (to prevent unnecessary string creation)

Reply:
Agreed. 
Thank you @abstractdog for the review !  :)